docs/docs/middleware.md

```diff
@@ -107,29 +108,35 @@ newService.orNotFound(goodRequest).unsafeRunSync()
 newService.orNotFound(badRequest).unsafeRunSync()
 ```
 
-It is possible for the wrapped `Service` to have different `Request` and `Response`
-types than the middleware. Authentication is, again, a good example. Authentication
-middleware is an `HttpService` (an alias for `Service[Request, Response]`) that wraps an `
-AuthedService` (an alias for `Service[AuthedRequest[T], Response]`. There is a type
-defined for this in the `http4s.server` package:
+Let's consider Authentication middleware as an example. Authentication
+middleware is a function that takes `AuthedRoutes[F]` 
+(an alias for `Kleisli[OptionT[F, *], AuthedRequest[F, T], Response[F]]`) 
```

Comment:
Instead of kleisli what about explaining it in terms of `AuthedRequest[F, T] => F[Option[Response[F]]]`

Reply:
good call, will fix this